src/poetry/core/factory.py

```diff
@@ -374,108 +527,257 @@ def create_dependency(
+                        " The latter will be ignored."
+                    )
+                elif deprecated:
+                    warning = (
+                        f"[tool.poetry.{name}] is deprecated."
+                        f" Use [project.{new_name}] instead."
+                    )
+                elif new_name not in dynamic:
+                    warning = (
+                        f"[tool.poetry.{name}] is set but '{new_name}' is not in"
+                        f" [project.dynamic]. If it is static use [project.{new_name}]."
+                        f" If it is dynamic, add '{new_name}' to [project.dynamic]."
+                    )
+                if warning:
+                    if additional_info := dynamic_information.get(name):
+                        warning += f"\n{additional_info}"
+                    result["warnings"].append(warning)
+
+        # scripts are special because entry-points are deprecated
+        # but files are not because there is no equivalent in [project]
+        if scripts := tool_poetry.get("scripts"):
+            for __, script in scripts.items():
+                if not isinstance(script, dict) or script.get("type") != "file":
+                    if "scripts" in project:
+                        warning = (
+                            "[project.scripts] is set and there are console scripts in"
+                            " [tool.poetry.scripts]. The latter will be ignored."
+                        )
+                    else:
+                        warning = (
+                            "Defining console scripts in [tool.poetry.scripts] is"
+                            " deprecated. Use [project.scripts] instead."
+                            " ([tool.poetry.scripts] should only be used for scripts"
+                            " of type 'file')."
                         )
+                    result["warnings"].append(warning)
+                    break
+
+        # dependencies are special because we consider
+        # [project.dependencies] as abstract dependencies for building
+        # and [tool.poetry.dependencies] as the concrete dependencies for locking
+        if (
+            tool_poetry["package-mode"]
+            and "dependencies" in tool_poetry
+            and "dependencies" not in project
+            and "dependencies" not in project.get("dynamic", [])
+        ):
+            result["warnings"].append(
+                "[tool.poetry.dependencies] is set but [project.dependencies] is not"
+                " and 'dependencies' is not in [project.dynamic]."
+                " You should either migrate [tool.poetry.depencencies] to"
+                " [project.dependencies] (if you do not need Poetry-specific features)"
+                " or add [project.dependencies] in addition to"
+                " [tool.poetry.dependencies] or add 'dependencies' to"
+                " [project.dynamic]."
+            )
```

Comment:
@radoering - I just want to confirm that if `project` exists, `project.dependencies` is omitted entirely, and `tool.poetry.dependencies` is used, `"dependencies"` must be included in `project.dynamic` (or Poetry will warn -- I believe that's what this warning is).

Reply:
(`pip-compile` and `uv` both now read `project.dependencies` directly if `project` exists and `"dependencies"` is not declared as "dynamic", so I just want to ensure that Poetry projects continue to work for those tools.)

Reply:
At the moment, there is an exception for non-package mode, but thinking about it again, I'll probably remove this exception, so yes:

When running `poetry check` and `project` exists and `"dependencies"` is not in `project.dynamic` and `tool.poetry.dependencies` exists, this warning will be printed.

Reply:
Cool, thanks. As long as users declare `"dependencies"` as `dynamic` in such cases, `uv` and `pip-compile` should continue to work fine for those projects. (I believe `pip` is also considering reading metadata directly from `pyproject.toml` in some cases so would apply there too.)